x/margin/types/add_or_append_coin.go

```diff
@@ -0,0 +1,14 @@
+package types
+
+import sdk "github.com/cosmos/cosmos-sdk/types"
+
+// Utility function to add a new coin or aggregate the amount if coin with same denom already exists
+func AddOrAppendCoin(coins []sdk.Coin, coin sdk.Coin) []sdk.Coin {
```

Comment:
I think this function is not required, `sdk.Coins(coins).Add(coin)` works exactly same as what this function does

Reply:
@jelysn not really the purpose of this function is to add an asset only if it does exist yet in `[]sdk.Coin`

Reply:
I don't see other functionality it provide other than sdk.Coins.Add function - can you check sdk.Coins.Add function and check difference?

Reply:
@jelysn updated the code to use `Coin.Add` good call out